### PR TITLE
PR for the branch (add-redux)

### PR DESCRIPTION
**Pull request by HamzaAnwar1998**
In this PR, I have created functionalities for adding and removing books. I have used redux to dispatch actions to reducer functions to achieve this

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
```
Comment (brytebee) [inline src/redux/books/books.js]: - [ ] **Required:** Your work is outstanding and works correctly. However, you are required to use the `ducks pattern` to define your action types. For example, your add action type could be re-written as `const ADD_BOOK = "bookstore/books/ADD_BOOK"`.
Kindly do this for `REMOVE_BOOK` as well and for every action type, you create going forward. This is to distinguish cases where different parts of your codebase have a similar action type. For example, if your `redux/books...` and `redux/categories...` all had an `ADD_BOOK` action type. I hope it makes sense. 😄

File: src/index.js [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
-  <React.StrictMode>
+  <Provider store={store}>
     <App />
-  </React.StrictMode>,
+  </Provider>,
 );
 
 // If you want to start measuring performance in your app, pass a function
```
Comment (brytebee) [inline src/index.js]: - [ ] **Required:** Kindly remove this dangling commented text from your codebase for a cleaner code base.

File: src/components/Books.js [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+// import React from 'react';
+import { useSelector } from 'react-redux';
 import Book from './Book';
 
-const Books = ({ books }) => books.map((book) => (
-  <Book book={book} key={book.id} />
-));
+// const Books = ({ books }) => books.map((book) => (
+//   <Book book={book} key={book.id} />
+// ));
```
Comment (brytebee) [inline src/components/Books.js]: - [ ] **Required:** Kindly clean up your codebase by removing the commented unused codes. This makes your work look more professional.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @HamzaAnwar1998👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- The project is structured correctly.✔️
- The app works fine.✔️
- The project documentation looks professional.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HamzaAnwar1998** [comment]
Thank you so much for allowing me to improve my code. Let me start working on it

**Review by julie-ify — APPROVED** [review]
Hi @HamzaAnwar1998, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**